central/policy/service/customresource/custom_resource.go

```diff
@@ -0,0 +1,78 @@
+package customresource
+
+import (
+	"bytes"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/protocompat"
+	"github.com/stackrox/rox/pkg/utils"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"gopkg.in/yaml.v3"
+)
+
+//
+//go:generate policy-as-code-helper-wrapper --type=storage.Policy
+//
+
+var (
+	invalidCRDNameCharPattern = regexp.MustCompile(`[^a-z0-9\.\-]`)
+	consecutivePattern        = regexp.MustCompile(`[.-]{2,}`)
+)
+
+// CustomResource represents the structure for a SecurityPolicy Kubernetes Custom Resource.
+type CustomResource struct {
+	APIVersion         string                 `yaml:"apiVersion"`
+	Kind               string                 `yaml:"kind"`
+	Metadata           map[string]interface{} `yaml:"metadata,omitempty"`
+	SecurityPolicySpec *Policy                `yaml:"spec"`
+}
+
+// GenerateCustomResource generate custom resource in YAML text from a policy
+func GenerateCustomResource(policy *storage.Policy) (string, error) {
+	convertedPolicy := ConvertPolicyToCustomResource(policy)
+	w := &bytes.Buffer{}
+	enc := yaml.NewEncoder(w)
+	defer utils.IgnoreError(enc.Close)
+	// Use idiomatic indentation.
+	enc.SetIndent(2)
+	if err := enc.Encode(convertedPolicy); err != nil {
+		return "", err
+	}
+	return w.String(), nil
+}
+
+func timestampToFormatRFC3339(ts *timestamppb.Timestamp) string {
+	t := protocompat.ConvertTimestampToTimeOrNil(ts)
+	if t == nil {
+		return ""
+	}
+	return t.Format(time.RFC3339)
+}
+
+// toDNSSubdomainName converts a policy name to a valid DNS subdomain name.
+func toDNSSubdomainName(name string) string {
+	name = strings.ToLower(name)
+	name = strings.ReplaceAll(name, " ", ".")
```

Comment:
My impression is that dashes are used much more often than dots in k8s names.

Reply:
Both dot and dashes are allowed. Dot are used more often in separating the field. I am ok with both.